### PR TITLE
[query] fix implementation of smartSummarize, add support for "median" as an aggregation func

### PR DESCRIPTION
**Pull request by teddywahle**
**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/query/graphite/ts/series.go [hunk 1]
```diff
@@ -598,6 +599,20 @@ func Max(a, b float64, count int) float64 { return math.Max(a, b) }
 // Last finds the latter of two values.
 func Last(a, b float64, count int) float64 { return b }
 
+// Median finds the median of a slice of values.
+func Median(vals []float64, count int) float64 {
+	if count == 1 {
```
Comment (arnikola) [inline src/query/graphite/ts/series.go:604]: Might be worth returning nan if count < 1

**Review by arnikola — APPROVED** [review]
LGTM with a nit about hardening an error scenario, again thanks for contributing!

File: src/query/graphite/ts/series.go [hunk 2]
```diff
@@ -598,6 +599,20 @@ func Max(a, b float64, count int) float64 { return math.Max(a, b) }
...
+		return vals[0]
```
Comment (teddywahle) [inline src/query/graphite/ts/series.go:605]:
```suggestion
func Median(vals []float64, count int) float64 {
        if count < 1 {
		return math.NaN()
        }
	if count == 1 {
		return vals[0]
```

**Review by arnikola — APPROVED** [review]
LGTM, thanks for contributing!

**Pull request merged**